assets/activation-scripts/activate.d/start.bash

```diff
@@ -0,0 +1,100 @@
+# If interactive and a command has not been passed, this is an interactive
+# activate,
+# and we print a message to the user
+# TODO: should this be printed after scripts?
+# Should it be in Rust using message::updated?
+if [ -t 1 ] && [ $# -eq 0 ]; then
+  echo "✅ You are now using the environment '$FLOX_ENV_DESCRIPTION'." >&2
+  echo "To stop using this environment, type 'exit'" >&2
+  echo >&2
+fi
+
+if [ $flox_env_found -eq 0 ]; then
+  # First activation of this environment. Snapshot environment to start.
+  _start_env="$($_coreutils/bin/mktemp --suffix=.start-env)"
+  export | $_coreutils/bin/sort > "$_start_env"
+else # we know "$_FLOX_ACTIVATE_FORCE_REACTIVATE" == true
+  # TODO: restore _start_env
+  :
+fi
+
+# Capture PID of this "first" activation. This provides the unique
+# identifier with which to refer to environment variables associated
+# with this environment activation.
+FLOX_ENV_PID="$$"
+
+# Set environment variables which represent the cumulative layering
+# of flox environments. For the most part this involves prepending
+# to the existing variables of the same name.
+# TODO: reconcile with CLI which should be setting these. Setting
+#       "*_activate" variables to indicate the ones we've seen and
+#       processed on the activate script side, and ultimately also
+#       for testing/comparison against the CLI-maintained equivalents.
+FLOX_ENV_DIRS_activate="$FLOX_ENV${FLOX_ENV_DIRS_activate:+:$FLOX_ENV_DIRS_activate}"
+FLOX_ENV_LIB_DIRS_activate="$FLOX_ENV/lib${FLOX_ENV_LIB_DIRS_activate:+:$FLOX_ENV_LIB_DIRS_activate}"
+FLOX_PROMPT_ENVIRONMENTS_activate="$FLOX_ENV_DESCRIPTION${FLOX_PROMPT_ENVIRONMENTS_activate:+ $FLOX_PROMPT_ENVIRONMENTS_activate}"
+export FLOX_ENV_DIRS_activate FLOX_ENV_LIB_DIRS_activate FLOX_PROMPT_ENVIRONMENTS_activate
+
+# Process the flox environment customizations, which includes (amongst
+# other things) prepending this environment's bin directory to the PATH.
+if [ -d "$FLOX_ENV/etc/profile.d" ]; then
+  declare -a _prof_scripts;
+  read -r -d '' -a _prof_scripts < <(
+    cd "$FLOX_ENV/etc/profile.d" || exit;
+    shopt -s nullglob;
+    echo *.sh;
+  );
+  # shellcheck disable=SC1090 # from rendered environment
+  for p in "${_prof_scripts[@]}"; do . "$FLOX_ENV/etc/profile.d/$p"; done
+  unset _prof_scripts;
```

Comment:
while we're here, how about
* renaming  `_prof_scripts` -> `_profile_scripts`
* use `source` instead of `.` (`.` is allegedly more posix, but the reset of the file a least is using `source`
* make that for loop multiline or run shfmt on this file

Reply:
Done